File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    while len(progresses)>0:
        count=0
        while len(progresses)>0 and progresses[0]>=100:
            count+=1
            progresses.pop(0)
            speeds.pop(0)
        for i in range(len(progresses)):
            progresses[i]=progresses[i]+speeds[i]
        if count>0:
            answer.append(count)
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int n=progresses.length;
        int[] days = new int [n];
        
        for (int i=0; i<n; i++){
            int remain = 100-progresses[i];
            days[i] = (remain+speeds[i]-1)/speeds[i];
        }
        
        List<Integer> answerList = new ArrayList<>();
        int maxDay = days[0];
        int count = 1;
        
        for (int i=1; i<n; i++){
            if(days[i]<=maxDay){
                count++;
            } else {
                answerList.add(count);
                maxDay = days[i];
                count = 1;
            }
        }
        answerList.add(count);
        
        int[] answer = new int[answerList.size()];
        for (int i=0; i<answerList.size(); i++){
            answer[i] = answerList.get(i);
        }
        return answer;
    }
}